modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -221,125 +220,116 @@ export const insertHtmlAtCaret = (editor: Editor, value: string, details: Insert
-  preProcessSetContent(editor, { content: value, format: 'html', set: false, selection: true, paste: details.paste }).each((args) => {
-    value = args.content;
+  // Add caret at end of contents if it's missing
+  if (value.indexOf('{$caret}') === -1) {
+    value += '{$caret}';
+  }
 
-    // Add caret at end of contents if it's missing
-    if (value.indexOf('{$caret}') === -1) {
-      value += '{$caret}';
+  // Replace the caret marker with a span bookmark element
+  value = value.replace(/\{\$caret\}/, bookmarkHtml);
+
+  // If selection is at <body>|<p></p> then move it into <body><p>|</p>
+  rng = selection.getRng();
+  const caretElement = rng.startContainer || (rng.parentElement ? rng.parentElement() : null);
+  const body = editor.getBody();
+  if (caretElement === body && selection.isCollapsed()) {
+    if (dom.isBlock(body.firstChild) && canHaveChildren(editor, body.firstChild) && dom.isEmpty(body.firstChild)) {
+      rng = dom.createRng();
+      rng.setStart(body.firstChild, 0);
+      rng.setEnd(body.firstChild, 0);
+      selection.setRng(rng);
     }
+  }
 
-    // Replace the caret marker with a span bookmark element
-    value = value.replace(/\{\$caret\}/, bookmarkHtml);
-
-    // If selection is at <body>|<p></p> then move it into <body><p>|</p>
-    rng = selection.getRng();
-    const caretElement = rng.startContainer || (rng.parentElement ? rng.parentElement() : null);
-    const body = editor.getBody();
-    if (caretElement === body && selection.isCollapsed()) {
-      if (dom.isBlock(body.firstChild) && canHaveChildren(editor, body.firstChild) && dom.isEmpty(body.firstChild)) {
-        rng = dom.createRng();
-        rng.setStart(body.firstChild, 0);
-        rng.setEnd(body.firstChild, 0);
-        selection.setRng(rng);
-      }
-    }
+  // Insert node maker where we will insert the new HTML and get it's parent
+  if (!selection.isCollapsed()) {
+    deleteSelectedContent(editor);
+  }
 
-    // Insert node maker where we will insert the new HTML and get it's parent
-    if (!selection.isCollapsed()) {
-      deleteSelectedContent(editor);
-    }
+  parentNode = selection.getNode();
 
-    parentNode = selection.getNode();
+  // Parse the fragment within the context of the parent node
+  const parserArgs: ParserArgs = { context: parentNode.nodeName.toLowerCase(), data: details.data, insert: true };
+  const fragment = parser.parse(value, parserArgs);
 
-    // Parse the fragment within the context of the parent node
-    const parserArgs: ParserArgs = { context: parentNode.nodeName.toLowerCase(), data: details.data, insert: true };
-    const fragment = parser.parse(value, parserArgs);
-
-    // Custom handling of lists
-    if (details.paste === true && InsertList.isListFragment(editor.schema, fragment) && InsertList.isParentBlockLi(dom, parentNode)) {
-      rng = InsertList.insertAtCaret(serializer, dom, selection.getRng(), fragment);
-      selection.setRng(rng);
-      editor.fire('SetContent', args);
```

Comment:
Was this here incorrectly? Or should it be replaced with firing an event somewhere else?

Reply:
It was moved to the `InsertContent.ts` file from what I saw, as it's handled by the `postProcessSetContent` call now I believe?

Reply:
Ah I didn't realise it was fired elsewhere already. It just seemed weird that the `fire` was removed without another added elsewhere in the PR.